tools/kubernetes/shared-cluster.sh

```diff
@@ -0,0 +1,149 @@
+        case $yn in
+            [yY]) ;;
+            [nN])
+                echo "exiting..."
+                exit 1
+                ;;
+            *) echo invalid response ;;
+        esac
+    fi
+    echo ""$ACTION" initiated on shared GKE cluster..."
+else
+    echo ""$ACTION" is not a valid action (only up, connect, or down)."
+    exit 1
+fi
+
+# Set a unique name for your cluster.
+export GCP_PROJECT_NAME=determined-ai
+export GKE_CLUSTER_NAME=backend-gke
+export GCS_BUCKET_NAME=backend-gke
+export GCS_SUBNET_NAME=backend-gke
+
+# CPU node configurations.
+export GKE_REGION=us-west1
+export GKE_NODE_LOCATION=us-west1-b
+export GKE_MACHINE_TYPE=n1-standard-8
+export GKE_NUM_NODES=1
+
+# GPU node pool configurations.
+export GKE_GPU_NODE_POOL_NAME=backend-gke
+export GKE_GPU_TYPE=nvidia-tesla-t4
+export GKE_GPU_PER_NODE=1
+
+# Bastion instance configuration.
+export BASTION_INSTANCE_NAME=backend-gke-bastion
+export BASTION_INSTANCE_ZONE="$GKE_REGION-b"
+export BASTION_INSTANCE_TYPE=e2-micro
+export BASTION_INSTANCE_STARTUP_HOOK=bastion-instance-startup.sh
+
+if [ ""$ACTION"" = "up" ]; then
+    gcloud container clusters create "$GKE_CLUSTER_NAME" \
+        --project "$GCP_PROJECT_NAME" \
+        --region "$GKE_REGION" \
+        --create-subnetwork=name="$GCS_SUBNET_NAME" \
+        --enable-master-authorized-networks \
+        --enable-ip-alias \
+        --enable-private-nodes \
+        --enable-private-endpoint \
+        --master-ipv4-cidr 172.16.0.32/28 \
+        --cluster-version=latest \
+        --node-locations "$GKE_NODE_LOCATION" \
+        --num-nodes="$GKE_NUM_NODES" \
+        --machine-type="$GKE_MACHINE_TYPE"
+
+    gcloud container node-pools create "$GKE_GPU_NODE_POOL_NAME" \
+        --cluster "$GKE_CLUSTER_NAME" \
+        --project "$GCP_PROJECT_NAME" \
+        --region="$GKE_REGION" \
+        --machine-type=n1-standard-8 \
+        --scopes=storage-full # --accelerator type="$GKE_GPU_TYPE",count="$GKE_GPU_PER_NODE"
+
+    gcloud compute instances create "$BASTION_INSTANCE_NAME" \
+        --project "$GCP_PROJECT_NAME" \
+        --zone="$BASTION_INSTANCE_ZONE" \
+        --machine-type="$BASTION_INSTANCE_TYPE" \
+        --network-interface=no-address,network-tier=PREMIUM,subnet="$GCS_SUBNET_NAME"
+
+elif [ ""$ACTION"" = "connect" ]; then
+    set -e
+    # Kill old tunnels.
+    # For some reason setting:
+    #   sudo sh -c 'echo "StreamLocalBindUnlink yes" >> /etc/ssh/sshd_config'
+    # on the server didn't work.
+    gcloud compute ssh "$BASTION_INSTANCE_NAME" \
+        --project "$GCP_PROJECT_NAME" \
+        --zone="$BASTION_INSTANCE_ZONE" \
+        --tunnel-through-iap \
+        -- pkill -u '$USER' -x -f '"^sshd: $USER[ ]*$"' || true
+
+    export MASTER_IP_FROM_INTERNAL=$(
+        gcloud compute instances describe "$BASTION_INSTANCE_NAME" \
+            --zone "$BASTION_INSTANCE_ZONE" \
+            --project "$GCP_PROJECT_NAME" \
+            | yq -o=json \
```

Comment:
Assuming this gets run on our local machines, when I first ran `./shared-cluster.sh connect`, I got a `yq: command not found` error and had to install with brew. Is there some way that we could ensure that `yq` and `jq` are already installed?